### DISCUSSION TITLE
Inquiry about the Technology Stack Used in the Project

### DISCUSSION THREAD

**Discussion by FruitTea2015** [open-webui/open-webui#1860] [Q&A]
Hello Open WebUI team and contributors,

I am very interested in the Open WebUI project and am considering contributing or integrating it into my own work. To better understand the project and prepare myself for development, I would like to know more about the technology stack that is currently being used.

Could you please provide information on:

The front-end frameworks and libraries (if any) that are utilized for the user interface.
The back-end frameworks and libraries that handle the server-side logic and API interactions.
Any other notable technologies or tools that are integral to the project.
Understanding these details will help me and potentially other contributors to get started more effectively.

Thank you for your time and assistance.

Best regards,
FruitTea2015

======================= Following are the Chinese version ======================

你好，Open WebUI团队和贡献者们，

我对Open WebUI项目非常感兴趣，并考虑为其贡献代码或将其集成到我的工作中。为了更好地理解项目并为开发做准备，我想知道项目目前所使用的技术栈。

能否提供以下信息：

用于用户界面的前端框架和库（如果有的话）。
处理服务器端逻辑和API交互的后端框架和库。
对项目至关重要的任何其他值得注意的技术或工具。
了解这些细节将有助于我和潜在的其他贡献者更有效地开始工作。

感谢你的时间和帮助。

此致，
FruitTea2015

**Comment by justinh-rahb** [comment]
Hi @FruitTea2015,

A quick glance at our repository and code should give you an idea of the technologies we're using. To summarize, our frontend is built with Svelte, while our backend relies on Python, primarily utilizing FastAPI.

**Reply by emilations** [reply]
Any reason why Svelte was chosen over more common tools?

**Reply by justinh-rahb** [reply]
I'll quote the author:
> i can write less and move faster
> something I'm just familiar with
> not a fan of react 😅
